src/randomeffectsterm.jl

```diff
@@ -23,9 +23,15 @@ function StatsModels.termvars(t::RandomEffectsTerm)
     vcat(StatsModels.termvars(t.lhs), StatsModels.termvars(t.rhs))
 end
 
-function StatsModels.apply_schema(t::FunctionTerm{typeof(|)}, schema::StatsModels.FullRank,
-        Mod::Type{<:MixedModel})
-    lhs, rhs = apply_schema.(t.args_parsed, Ref(schema), Mod)
+function StatsModels.apply_schema(t::FunctionTerm{typeof(|)},
+                                  schema::StatsModels.FullRank,
+                                  Mod::Type{<:MixedModel})
+    schema = StatsModels.FullRank(schema.schema)
+    lhs, rhs = t.args_parsed
+    if !StatsModels.hasintercept(lhs) && !StatsModels.omitsintercept(lhs)
```

Comment:
I understand the logic of this line (neither explicit "commission" nor explicit omission), but the surface appearance is "if both mutually exclusive conditions hold" .... (no change necessary)